### ISSUE TITLE
Gradle nativeBuild does not with Bellsoft Liberica NIK SDK

### ISSUE THREAD

**Issue by lukas-krecan** [spring-projects-experimental/spring-native#1381]
The documentation [recommends](https://docs.spring.io/spring-native/docs/current/reference/htmlsingle/#getting-started-native-image-system-requirements) to use Bellsoft Liberica NIK SDK but if you run `./gradlew nativeBuild` you get
```
 No compatible toolchains found for request filter: {languageVersion=17, vendor=matching('GraalVM'), implementation=vendor-specific} (auto-detect true, auto-download true)
```
If I switch to grl SDK, it works

It would be nice to either fix the docs or the build.

**Comment by sdeleuze** [comment]
I tried and it works for me. Please make sure you ran `gu install native-image` with Liberica NIK before since this is a per SDK command to run. Side note : `nativeBuild` is now deprecated in favor of `nativeCompile`.

Please let me know how it goes.

**Comment by lukas-krecan** [comment]
I did `gu install native-image` (I spent around an hour trying fixing it). Can you try to uninstall the `grl`? My suspicion is that Gradle looks for `GraalVM` vendor and if you have it installed, it just finds it.

**Comment by sdeleuze** [comment]
Yes I will, maybe the is something specific we should fine tune on https://github.com/graalvm/native-build-tools side.

**Comment by sdeleuze** [comment]
Ok so short term we are going to disable toolchain support as implemented via https://github.com/graalvm/native-build-tools/issues/164 and document that as part of Spring Native `0.11.1`, see [related documentation](https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html#configuration-toolchains-disabling).

Middle term, Gradle Toolchain support should be made more powerful to support `native-image` flavors of Java, you can follow https://github.com/gradle/gradle/issues/18896 for more details. https://github.com/gradle/gradle/pull/18028 is also kind of related.

**Comment by sdeleuze** [comment]
At the moment, the documented workaround does not seems to work:
```
tasks.withType(org.graalvm.buildtools.gradle.tasks.BuildNativeImageTask).configureEach {
	disableToolchainDetection = true
}
```
I have asked NBT team feedback on this and will keep you informed.

**Comment by sdeleuze** [comment]
I think I understand why it does not work, and it looks like a NBT 0.9.8 bug from what I can say. I created https://github.com/graalvm/native-build-tools/issues/183 and will try to see how fast we can get NBT 0.9.9 to fix it.

For now, with Gradle and Native Build Tools use GraalVM CE.

**Comment by sdeleuze** [comment]
The bug is now confirmed and fixed in Native Build Tools `0.9.9-SNAPSHOT`.

I am waiting to know if a quick NBT `0.9.9` release is possible, in that case, we will disable by default Gradle toolchain for native compilation.

Otherwise, I will move that issue to Spring Native `0.11.2` milestone and document in `0.11.1` that Liberica NIK is not yet supported with NBT + Gradle.
